CMakeLists.txt

```diff
@@ -39,6 +39,33 @@ project(
     LANGUAGES
         CXX
 )
+
+# Adjust compiler and linker flags needed by this project
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    find_library(LIBC++ c++)
+    find_library(LIBC++ABI c++abi)
+    if(NOT LIBC++ OR NOT LIBC++ABI)
+        message(
+            FATAL_ERROR
+            "libc++ or libc++abi not found. Make sure you have libc++ and libc++abi installed and in your PATH"
+        )
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
```

Comment:
Pretty sure we'd be better served with `add_compiler_options` and `add_link_options` to set it at the directory scope and not worry about cache vars (which I think already bit you while refactoring this).